csrc/scheduler/registry.cpp

```diff
@@ -1335,11 +1329,36 @@ class NoOpScheduler : public SchedulerEntry {
       HeuristicSummary* data_cache = nullptr) {
     // TODO:
     //  Pipe through dynamic zero checks.
+
+    for (auto output : ir_utils::filterByType<TensorView>(fusion->outputs())) {
+      for (auto id : output->domain()->root()) {
+        // Check that there are no concrete output dimensions of non-zero size
+        if (id->isReduction() || id->isBroadcast()) {
+          continue;
+        }
+        if (runtime_info.expressionEvaluator()
+                .evaluate(id->extent())
+                .value()
+                .as<int64_t>() != 0) {
+          return false;
+        }
+      }
+    }
+
     return true;
   }
 
   void schedule(Fusion* fusion) override {
-    // Schedule is no-op.
+    // Schedule is no-op, but we need to remove problematic ops. In particular,
+    // reductions with zero concrete size should be converted to full ops.
+    fusion->print();
+    std::cout << "\nRemoving reductions\n" << std::endl;
+    for (auto rop : fusion->exprs()) {
+      if (rop->isA<ReductionOp>() || rop->isA<WelfordOp>()) {
+        ir_utils::replaceReductionWithFull(rop);
```

Comment:
When would this get hit? Why do we need to remove the reduction ops in a no schedule situation? Is this because of some runtime check?